R/derive_joined.R

```diff
@@ -399,7 +399,7 @@ derive_vars_joined <- function(dataset,
         suffix = ".join"
       ),
       mode = mode,
-      check_type = check_type
+      check_type = "none"
```

Comment:
@bundfussr This modification addresses the get_duplicates reprex you cited in the issue, though I'm not sure this is the "perfect" solution to it. There are 3 places where `check_type` for the duplicates are being checked, up at line 364 (which is set to "none"), here in line 402 (which is passed through the user in the function call), and passively through `derive_vars_merged()` in line 413 (which is defaulted to "warning"). 

May be important to note, that in our vignette `generic.Rmd`, we have `check_type = "none"` in one of the examples to suppress what would've been a similar warning you have in your reprex, because the default for `check_type` right now in `derive_vars_joined()` is `"warning"`.


![image](https://github.com/pharmaverse/admiral/assets/43018762/9222da78-a650-4f39-9994-19dc5f062015)

Reply:
This line should not be changed. Otherwise, no message is issued if the selected observation is not unique. Consider for example
```
derive_vars_joined(
  adlb_ast,
  dataset_add = adlb_tbili_pbl,
  by_vars = exprs(STUDYID, USUBJID),
  order = exprs(ADT),
  new_vars = exprs(TBILI_ADT = ADT, TBILI_SEQ = ASEQ),
  filter_join = ADT <= ADT.join,
  mode = "first"
)
```
with the datasets from the second example in the issue. Here a warning should be issued because the first TBILI assessment at or after the AST assessment is not unique for `ASEQ %in% c(2, 3)`.

Reply:
If thats the case, is the example in the `generic.Rmd` a little misleading then? If it weren't toggled to `check_type = "none"` there would be a warning. I also think there is some overall confusion with how `order` is handled in general which might be the source of the troubles here

![image](https://github.com/pharmaverse/admiral/assets/43018762/61e29c92-8dfa-4f14-a7f5-70150e7a5040)

In general, `!!!chr2vars(names(order))` does **not** select columns so the `common_vars` statement a little further down can be inconsistent, depending if the user plugged in a function and used a var-val declaration e.g `convert_dtc_to_dt()` inside order

Reply:
Specifying `check_type = "none"` in the example makes sense because only the maximum severity is added as new variable. If there are multiple observations with the maximum severity, it doesn't matter which one is used.

Regarding `order`, we need to more comments to the code. The purpose of the `mutate()` call is to create variables for named expressions in `order`. Then these can be used in `join_vars` or the filter condition.
You are right that `!!!chr2vars(names(order))` is not correct. We may lose the variables used in the expressions of `order`. I think it should be replaced by `!!!replace_values_by_names(extract_vars(order))`.